src/rgw/rgw_op.cc

```diff
@@ -5828,6 +5831,11 @@ void RGWSetRequestPayment::execute(optional_yield y)
 
 int RGWInitMultipart::verify_permission(optional_yield y)
 {
+  #ifdef HAVE_JAEGER
+  auto verify_permission_span = rgw_tracer->start_span("verify_permission", this->trace);
+  verify_permission_span->SetTag(tracing::USER_DISPLAY_NAME, s->user->get_display_name());
+  verify_permission_span->SetTag(tracing::BUCKET_NAME, s->bucket->get_name());
+  #endif
```

Comment:
i'd really prefer not to have these `#ifdefs` all over. to get rid of them, we have a couple options:

1. require `WITH_JAEGER` for rgw by disabling `WITH_RADOSGW` when `WITH_JAEGER` is disabled
2. add stubs that provide the same interface when `HAVE_JAEGER` isn't defined

i think option 1. is reasonable because jaeger and opentracing are included as submodules of the ceph tree, so `WITH_JAEGER` will always be enabled for official ceph builds, and other distros won't have issues with missing packages

option 2. is what we chose for the blkin tracing project (see `common/zipkin_trace.h`). the potential advantage here is that we could run benchmarks in 3 different configurations: 2a) `WITH_JAEGER` disabled, 2b) `WITH_JAEGER` enabled but runtime-disabled, and 2c) with tracing enabled

Reply:
lets go with option (2)